### ISSUE TITLE
Unclear Error

### ISSUE THREAD

**Issue by illuzen** [trufflesuite/truffle-compile#4]
I write an abstract contract interface like so:

`contract MyContract {
     function foo() public {
}`

Uh, oh! Typo! Fortunately the compiler will catch it for me...

/usr/local/lib/node_modules/truffle/node_modules/truffle-compile/profiler.js:120
        if (ancestors.length > 0) {
                     ^

TypeError: Cannot read property 'length' of undefined
    at walk_from (/usr/local/lib/node_modules/truffle/node_modules/truffle-compile/profiler.js:120:22)
    at Array.forEach (native)
    at /usr/local/lib/node_modules/truffle/node_modules/truffle-compile/profiler.js:129:13
    at /usr/local/lib/node_modules/truffle/node_modules/truffle-compile/profiler.js:228:7
    at /usr/local/lib/node_modules/truffle/node_modules/truffle-compile/node_modules/async/dist/async.js:843:16
    at /usr/local/lib/node_modules/truffle/node_modules/truffle-compile/node_modules/async/dist/async.js:4956:25
    at apply (/usr/local/lib/node_modules/truffle/node_modules/truffle-compile/node_modules/async/dist/async.js:21:25)
    at /usr/local/lib/node_modules/truffle/node_modules/truffle-compile/node_modules/async/dist/async.js:56:12
    at /usr/local/lib/node_modules/truffle/node_modules/truffle-compile/profiler.js:191:18
    at /usr/local/lib/node_modules/truffle/node_modules/truffle-resolver/index.js:77:5
    at /usr/local/lib/node_modules/truffle/node_modules/truffle-resolver/node_modules/async/internal/onlyOnce.js:12:16
    at /usr/local/lib/node_modules/truffle/node_modules/truffle-resolver/node_modules/async/whilst.js:63:18
    at apply (/usr/local/lib/node_modules/truffle/node_modules/lodash/_apply.js:15:25)
    at /usr/local/lib/node_modules/truffle/node_modules/lodash/_overRest.js:32:12
    at /usr/local/lib/node_modules/truffle/node_modules/truffle-resolver/index.js:62:7
    at /usr/local/lib/node_modules/truffle/node_modules/truffle-resolver/fs.js:41:12

waht

**Comment by adklempner** [comment]
I received the same error when I declared a state variable outside of the contract declaration
```
pragma solidity ^0.4.4;

import "./OtherContract.sol";

OtherContract other;

contract MyContract {
   function MyContract() {
     other = new OtherContract();
   }
}
```

**Comment by recmo** [comment]
Just got hit by the same problem. Much confusion ensued. I was hoping syntax errors would result in nice `file:line: message` output.

**Comment by tcoulter** [comment]
This appears to be fixed in the latest version of Truffle (v4.0.0-beta.0). Here's the solidity contract I used:

```
pragma solidity ^0.4.4;

contract MyContract { function foo() public { }
``` 

The error message I receive: 

```
$ truffle compile
SyntaxError: Expected "{", "}", any character, comment, end of line, string, or whitespace but end of input found.
    at peg$buildStructuredError (C:\Users\Tim\AppData\Roaming\nvm\v8.4.0\node_modules\truffle\build\cli.bundled.js:295340:12)
    at Object.peg$parse [as parse] (C:\Users\Tim\AppData\Roaming\nvm\v8.4.0\node_modules\truffle\build\cli.bundled.js:299020:11)
    at Object.parse (C:\Users\Tim\AppData\Roaming\nvm\v8.4.0\node_modules\truffle\build\cli.bundled.js:313578:23)
    at C:\Users\Tim\AppData\Roaming\nvm\v8.4.0\node_modules\truffle\build\cli.bundled.js:69422:36
    at C:\Users\Tim\AppData\Roaming\nvm\v8.4.0\node_modules\truffle\build\cli.bundled.js:58940:5
    at C:\Users\Tim\AppData\Roaming\nvm\v8.4.0\node_modules\truffle\build\cli.bundled.js:78300:16
    at next (C:\Users\Tim\AppData\Roaming\nvm\v8.4.0\node_modules\truffle\build\cli.bundled.js:88307:18)
    at C:\Users\Tim\AppData\Roaming\nvm\v8.4.0\node_modules\truffle\build\cli.bundled.js:58925:7
    at C:\Users\Tim\AppData\Roaming\nvm\v8.4.0\node_modules\truffle\build\cli.bundled.js:200901:5
    at C:\Users\Tim\AppData\Roaming\nvm\v8.4.0\node_modules\truffle\build\cli.bundled.js:88212:16
```

Going to close this issue. Please reopen if you can pin it down to a more specific configuration, or feel free to open a new issue. Cheers!
